lib/completion.dart

```diff
@@ -71,8 +72,8 @@ class DartCompleter extends CodeCompleter {
         // Filter unmatching completions.
         // TODO: This is temporary; tracking issue here:
         // https://github.com/dart-lang/dart-services/issues/87.
-        if (delta > 0) {
-          if (!completion.text.toLowerCase().startsWith(lowerPrefix)) {
+        if (lowercaseReplacementString.isNotEmpty) {
+          if (!completion.text.toLowerCase().contains(lowercaseReplacementString)) {
```

Comment:
You could add a method on `Completion`, `bool matches(String)`, that would do a lowercase and a contains.